roles/matrix-ntfy/templates/ntfy/server.yml.j2

```
@@ -1,3 +1,11 @@
 base_url: {{ matrix_ntfy_base_url }}
 behind_proxy: true
 cache_file: /data/cache.db
+
+# Rate Limits
+global-topic-limit: {{ matrix_ntfy_limit_global_topic }}
+visitor-subscription-limit: {{ matrix_ntfy_limit_visitor_subscriptions }}
+
+visitor-request-limit-burst: {{ matrix_ntfy_limit_burst_rate }}
+visitor-request-limit-replenish: "{{ matrix_ntfy_limit_replenish_rate }}"
+visitor-request-limit-exempt-hosts: "{{ matrix_ntfy_limit_rate_exempt }}"
```

```suggestion
global-topic-limit: {{ matrix_ntfy_limit_global_topic | to_json }}
visitor-subscription-limit: {{ matrix_ntfy_limit_visitor_subscriptions | to_json }}

visitor-request-limit-burst: {{ matrix_ntfy_limit_burst_rate | to_json }}
visitor-request-limit-replenish: "{{ matrix_ntfy_limit_replenish_rate | to_json }}"
visitor-request-limit-exempt-hosts: "{{ matrix_ntfy_limit_rate_exempt | to_json }}"
```

It's always more reliable to use ` | to_json` in YAML/JSON config files.

-----

However, another thing that stands out is: variables are named differently than what they control. We usually strive for a match here, as it's more straightforward. A setting called `global-topic-limit` should be controlled via a variable called `matrix_ntfy_global_topic_limit`. Renaming all these variables will make this more consistent and straightforward.

---

will do

---

For some reason, when I `to_json` the last two, they break with
```
fatal: [matrix.matrix753.malhotra.cc]: FAILED! => 
  msg: |-
    An unhandled exception occurred while templating '{{ matrix_ntfy_configuration_yaml | from_yaml | combine(matrix_ntfy_configuration_extension, recursive=True) }}'. Error was a <class 'yaml.parser.ParserError'>, original message: while parsing a block mapping
      in "<unicode string>", line 1, column 1
    did not find expected key
      in "<unicode string>", line 11, column 39
```

---

Oh, sorry.. When you use `| to_json`, you shouldn't add double-quotes around it:

```diff
-"{{ something | to _json }}"
+{{ something | to_json }}
```

My suggested code above is wrong.